opennms-services/src/main/java/org/opennms/netmgt/poller/QueryManagerDaoImpl.java

```diff
@@ -318,4 +318,33 @@ public void updateServiceStatus(int nodeId, String ipAddr, String serviceName, S
                     serviceName, nodeId,  ipAddr, status, e);
         }
     }
+    
+    @Override
+    public void updateServiceLastGood(int nodeId, String ipAddr, String serviceName, Date updateTime) {
+        try {
+            OnmsMonitoredService service = m_monitoredServiceDao.get(nodeId, InetAddress.getByName(ipAddr), serviceName);
+            service.setLastGood(updateTime);;
```

Comment:
Is updating the lastGood and lastFail columns only informational, or do they have some formal purpose?

I noticed the columns were already in the database, but appeared to be unused prior to this.